lib/tools.py

```diff
@@ -56,32 +57,16 @@ def dt2ts(self, dt):
         return time.mktime(dt.timetuple())
 
     def fetch_url(self, url, username=None, password=None, timeout=2):
-        headers = {'Accept': 'text/plain'}
-        plain = True
-        if url.startswith('https'):
-            plain = False
-        lurl = url.split('/')
-        host = lurl[2]
-        purl = '/' + '/'.join(lurl[3:])
-        if plain:
-            conn = http.client.HTTPConnection(host, timeout=timeout)
-        else:
-            conn = http.client.HTTPSConnection(host, timeout=timeout)
-        if username and password:
-            headers['Authorization'] = ('Basic '.encode() + base64.b64encode((username + ':' + password).encode()))
-        try:
-            conn.request("GET", purl, headers=headers)
-        except Exception as e:
-            logger.warning("Problem fetching {0}: {1}".format(url, e))
-            conn.close()
-            return False
-        resp = conn.getresponse()
+        req = urllib.request.Request(url)
```

Comment:
What about the username and password parameters? They are not included in the request in any way.